### PR TITLE
Implement graphql mutation node for delete lead in bulk 

### PR DESCRIPTION
**Pull request by susilnem**
Addresses 
-  https://github.com/the-deep/server/issues/1459

## Changes
-  Add mutation for bulk delete
- Generate schema
- Add test cases for all permission

Mention related users here if any.

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [x] permission checks (tests here too)
- [x] translations

### CODE REVIEW

**Comment by thenav56** [comment]
@susilnem, I  think this is almost already done, We just need to add a new line and some test cases.

https://github.com/the-deep/server/blob/develop/apps/lead/mutation.py#L92-L101
https://github.com/the-deep/server/blob/develop/apps/entry/mutation.py#L69

File: apps/lead/tests/test_mutations.py [hunk 1]
```diff
@@ -389,6 +389,45 @@ def _query_check(**kwargs):
...
+        # Error without login
+        _query_check(assert_for_error=True)
+
+        # --- login
+        self.force_login(user)
+        self.assertEqual(lead_count, Lead.objects.count())
+        # Success with normal lead (with project membership)
+        result = _query_check()['data']['project']['leadBulk']
+        self.assertMatchSnapshot(result, 'success')
+        self.assertEqual(lead_count - 2, Lead.objects.count())
```
Comment (k9845) [inline apps/lead/tests/test_mutations.py]: Make sure to add test for different user with different permission.
Also Instead of count, let's assert for id.

**Review by k9845 — CHANGES_REQUESTED** [review]

File: apps/lead/tests/test_mutations.py [hunk 2]
```diff
@@ -389,6 +389,62 @@ def _query_check(**kwargs):
...
+        # -- With login (non-member)
+        self.force_login(non_member_user)
+        _query_check(assert_for_error=True)
+
+        # --- login (member)
+        self.force_login(member_user)
+        self.assertEqual(lead_count, Lead.objects.count())
+        # Success with normal lead (with project membership)
+        result = _query_check()['data']['project']['leadBulk']
+        self.assertMatchSnapshot(result, 'success')
```
Comment (thenav56) [inline apps/lead/tests/test_mutations.py]: This is not required (same as next line)
> NOTE: For simple field check, we don't need snapshot test.

File: apps/lead/tests/test_mutations.py [hunk 3]
```diff
@@ -389,6 +389,62 @@ def _query_check(**kwargs):
...
+        self.assertEqual(
+            result,
+            {
+                'errors': [],
+                'deletedResult': [
+                    dict(title=lead1.title),
+                    dict(title=lead2.title),
```
Comment (thenav56) [inline apps/lead/tests/test_mutations.py]: Let's add id as well

File: apps/lead/tests/test_mutations.py [hunk 4]
```diff
@@ -389,6 +389,62 @@ def _query_check(**kwargs):
...
+                        title
+                    }
+                }
+              }
+            }
+        '''
+
+        project = ProjectFactory.create()
+        non_member_user = UserFactory.create()
+        member_user = UserFactory.create()
```
Comment (thenav56) [inline apps/lead/tests/test_mutations.py:410]: Let's add readonly_member_user as well.

File: apps/lead/tests/test_mutations.py [hunk 5]
```diff
@@ -389,6 +389,62 @@ def _query_check(**kwargs):
...
+        # Error without login
+        _query_check(assert_for_error=True)
+
+        # -- With login (non-member)
+        self.force_login(non_member_user)
+        _query_check(assert_for_error=True)
+
+        # --- login (member)
+        self.force_login(member_user)
+        self.assertEqual(lead_count, Lead.objects.count())
```
Comment (thenav56) [inline apps/lead/tests/test_mutations.py]: Let's do this after each _query_check

**Review by thenav56 — CHANGES_REQUESTED** [review]
Looks good. Let's add more cases to the test.

File: apps/lead/tests/test_mutations.py [hunk 6]
```diff
@@ -389,6 +389,71 @@ def _query_check(**kwargs):
...
+
+        # --- login as read-only member
+        self.force_login(read_only_member_user)
+        _query_check(assert_for_error=True)
+        self.assertEqual(lead_count, Lead.objects.count())
+
+        # --- login (member)
+        self.force_login(member_user)
+        self.assertEqual(lead_count, Lead.objects.count())
```
Comment (thenav56) [inline apps/lead/tests/test_mutations.py]:
```suggestion
```

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: apps/lead/tests/test_mutations.py [hunk 7]
```diff
@@ -389,6 +389,69 @@ def _query_check(**kwargs):
...
+        self.assertEqual(
+            result,
+            {
+                'errors': [],
+                'deletedResult': [
+                    dict(id=str(lead1.pk), title=lead1.title),
+                    dict(id=str(lead2.pk), title=lead2.title),
+                ],
+            },
+        )
```
Comment (thenav56) [inline apps/lead/tests/test_mutations.py:452]: Let's add a final
```suggestion
        )
        self.assertEqual(0, Lead.objects.count())
```

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: apps/lead/tests/test_mutations.py [hunk 8]
```diff
@@ -389,6 +389,69 @@ def _query_check(**kwargs):
...
+
+        project = ProjectFactory.create()
+        non_member_user = UserFactory.create()
+        read_only_member_user = UserFactory.create()
+        member_user = UserFactory.create()
+        project.add_member(member_user, role=self.project_role_member)
+        project.add_member(read_only_member_user, role=self.project_role_reader_non_confidential)
+
+        lead1 = LeadFactory.create(project=project)
+        lead2 = LeadFactory.create(project=project)
```
Comment (thenav56) [inline apps/lead/tests/test_mutations.py]: Add other nose lead

File: apps/lead/tests/test_mutations.py [hunk 9]
```diff
@@ -389,6 +389,69 @@ def _query_check(**kwargs):
...
+        '''
+
+        project = ProjectFactory.create()
+        non_member_user = UserFactory.create()
+        read_only_member_user = UserFactory.create()
+        member_user = UserFactory.create()
+        project.add_member(member_user, role=self.project_role_member)
+        project.add_member(read_only_member_user, role=self.project_role_reader_non_confidential)
+
+        lead1 = LeadFactory.create(project=project)
```
Comment (thenav56) [inline apps/lead/tests/test_mutations.py]: Use create_batch batch_create. use 3

**Review by thenav56 — DISMISSED** [review]

**Review by sudan45 — DISMISSED** [review]

**Pull request merged**